contracts/permissionless/OrderList.sol

```diff
@@ -281,7 +281,7 @@ contract OrderList is PermissionGroups, OrderListInterface {
         Order storage order = orders[orderId];
 
         // Make sure such order exists in mapping.
-        require(order.prevId != 0 || order.nextId != 0);
+        require(order.prevId != 0 && order.nextId != 0 && order.prevId != TAIL_ID && order.nextId != HEAD_ID);
```

Comment:
Consider splitting the two situations: 
- Order was never in the list (order.prevId == 0 or order.nextId == 0), and 
- Order was already removed (prevId == TAIL or nextId == HEAD)